pyanaconda/modules/subscription/rhsm_observer.py

```diff
@@ -0,0 +1,73 @@
+#
+# Copyright (C) 2020  Red Hat, Inc.  All rights reserved.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+from pyanaconda.modules.common.constants.services import RHSM
+
+from pyanaconda.anaconda_loggers import get_module_logger
+
+from dasbus.client.observer import DBusObserver, DBusObserverError
+
+log = get_module_logger(__name__)
+
+# How long to wait for a systemd unit backed DBus service
+# to become available (in seconds). This timeout is based
+# on the default 90 second timeout for manual systemd unit
+# activation, which is also 90 seconds.
+SERVICE_ACTIVATION_TIMEOUT = 90.0
+
+class RHSMObserver(DBusObserver):
+    """Observer of the systemd unit backed RHSM DBus service."""
+
+    def __init__(self, message_bus, startup_check_method, timeout=SERVICE_ACTIVATION_TIMEOUT):
+        """Creates a RHSM service observer.
+
+        The observer will wait (up to a timeout) for the task starting the RHSM service
+        to finish and then provide access to the corresponding RHSM DBus API.
+
+        The waiting is only expected to happen at Anaconda startup, once the RHSM
+        startup task has finished running no further waiting is expected.
+
+        :param startup_check_method: a blocking method that waits for RHSM
+                                     startup to finish up to a given timeout
+                                     and then returns a boolean value corresponding
+                                     to True if startup was successful or False
+                                     otherwise
+        :param float timeout: how long to wait for RHSM service to start in seconds
+        """
+        super().__init__(message_bus, RHSM.service_name)
+        self._startup_check_method = startup_check_method
+        self._timeout = timeout
+
+    def get_proxy(self, object_identifier):
```

Comment:
The argument `object_identifier` could be None by default, if you need to be able to access also `RHSM.get_proxy()`.

Reply:
I've checked the code on RHEL-8 and it looks like we don't do that, so I did I like this.